downloads/swan-lake-release-notes/swan-lake-preview8/RELEASE_NOTE.md

```diff
@@ -0,0 +1,519 @@
+- The resource functions are changed to remote functions in the new listener APIs.
+- The service name is given as a string with the new Ballerina language changes.
+- The `onMessage` method of the `email:Listener` (i.e., the new `email:PopListener` and `email:ImapListener`) are changed to `onEmailMessage`.
+- The `pollingInterval` field of the `email:Listener` is changed to `pollingIntervalInMillis` in new listener APIs. That makes it consistent across the other Ballerina modules, which are time durations configured in milliseconds.
+
+A sample POP3 listener is given below.
+
+**Old Syntax**
+
+```ballerina
+email:PopConfig popConfig = {
+     port: 995,
+     enableSsl: true
+};
+
+listener email:Listener emailListener = new ({
+    host: "pop.email.com",
+    username: "reader@email.com",
+    password: "pass456",
+    protocol: "POP",
+    protocolConfig: popConfig,
+    pollingInterval: 2000
+});
+
+service emailObserver on emailListener {
+
+    resource function onMessage(email:Email emailMessage) {
+
+    }
+
+    resource function onError(email:Error emailError) {
+
+    }
+
+}
+```
+
+**New Syntax**
+
+```ballerina
+listener email:PopListener emailListener = new ({
+    host: "pop.email.com",
+    username: "reader@email.com",
+    password: "pass456",
+    pollingIntervalInMillis: 2000,
+    port: 995,
+    enableSsl: true
+});
+
+service "emailObserver" on emailListener {
+
+    remote function onEmailMessage(email:Message emailMessage) {
+
+    }
+
+    remote function onError(email:Error emailError) {
+
+    }
+
+}
+```
+
+##### WebSub Module Changes
+
+- The base path is removed from the `SubscriberServiceConfig`.
+- The `onNotification` and `onIntentVerification` resources are converted to remote functions.
+
+**Old Syntax**
+
+```ballerina
+@websub:SubscriberServiceConfig {
+  path: "/websub",
+  target: ["http://localhost:9191/websub/hub", "http://websubpubtopic.com"]
+}
+service websubSubscriber on new websub:Listener(8181) {
+  resource function onIntentVerification(websub:Caller caller,  websub:IntentVerificationRequest request) {}
+  resource function onNotification(websub:Notification notification) {}
+}
+```
+
+**New Syntax**
+
+```ballerina
+@websub:SubscriberServiceConfig {
+   target: ["http://localhost:23191/websub/hub", "http://one.websub.topic.com"]
+}
+service websub:SubscriberService /websub on new websub:Listener(8181) {
+   remote function onIntentVerification(websub:Caller caller, websub:IntentVerificationRequest request) {}
+   remote function onNotification (websub:Notification notification) {}
+}
+```
+
+##### UUID Module Changes
```

Comment:
UUID is a new module introduced in SLP8.

Reply:
Fixed.